x-pack/plugin/ilm/src/internalClusterTest/java/org/elasticsearch/xpack/ilm/ILMHistoryTests.java

```diff
@@ -91,8 +91,12 @@ public void testIlmHistoryIndexCanRollover() throws Exception {
 
         //wait for all history items to index to avoid waiting for timeout in ILMHistoryStore beforeBulk
         assertBusy(() -> {
-            SearchResponse search = client().prepareSearch(firstIndex).setQuery(matchQuery("index", firstIndex)).setSize(0).get();
-            assertThat(search.getHits().getTotalHits().value, is(9L));
+            try {
+                SearchResponse search = client().prepareSearch(firstIndex).setQuery(matchQuery("index", firstIndex)).setSize(0).get();
+                assertThat(search.getHits().getTotalHits().value, is(9L));
+            } catch (Exception e) {
+                fail(e.getMessage());
```

Comment:
Thanks for fixing this Przemko.

Can you add a comment on why this is needed please? it's not obvious `assertBusy` fails on first non-AssertionError exception. We could also use `assertHitCount` maybe? 

Is it worth also increasing the timeout of `assertBusy` to 30 seconds?

Reply:
I've added a comment and switched to `assertHitCount`. I don't think we need to increase timeout - we should have to wait for at most 5s for all ILM history items to be index